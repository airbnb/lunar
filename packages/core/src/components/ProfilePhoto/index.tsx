import React from 'react';
import PropTypes from 'prop-types';
import { and, mutuallyExclusiveProps } from 'airbnb-prop-types';
import withStyles, { WithStylesProps } from '../../composers/withStyles';

const mutuallyExclusiveSizePropType = mutuallyExclusiveProps(
  PropTypes.any,
  'small',
  'large',
  'macro',
  'size',
);
const namedSizePropType = and([PropTypes.bool, mutuallyExclusiveSizePropType]);
const unitSizePropType = and([PropTypes.number, mutuallyExclusiveSizePropType]);

export type Props = {
  /** URL of the image to display. */
  imageSrc: string;
  /** Whether the component should be inline. */
  inline?: boolean;
  /** Large size (96px). */
  large?: boolean;
  /** Macro size (160px). */
  macro?: boolean;
  /** Size in units to multiply by. */
  size?: number;
  /** Small size (24px). */
  small?: boolean;
  /** Whether the image should be square (it is rounded by default). */
  square?: boolean;
  /** Accessibility text for the Photo. */
  title: string;
};

/** Display a profile photo. */
export class ProfilePhoto extends React.Component<Props & WithStylesProps> {
  static propTypes = {
    large: namedSizePropType,
    macro: namedSizePropType,
    size: unitSizePropType,
    small: namedSizePropType,
  };

  // Do not supply defaults for the mutually exclusive props,
  // doing so will cause validation errors.
  static defaultProps = {
    inline: false,
    square: false,
  };

  render() {
    const {
      cx,
      imageSrc,
      inline,
      macro,
      large,
      small,
      size,
      square,
      styles,
      title,
      theme,
    } = this.props;
    const { unit } = theme!;

    return (
      <div
        className={cx(
          inline && styles.inline,
          styles.regular,
          macro && styles.macro,
          large && styles.large,
          small && styles.small,
          !!size &&
            size > 0 && {
              height: size * unit,
              maxHeight: size * unit,
              maxWidth: size * unit,
              width: size * unit,
            },
        )}
      >
        <img
          className={cx(
            styles.image,
            styles.regular,
            !square && styles.roundedImage,
            macro && styles.macro,
            large && styles.large,
            small && styles.small,
            !!size &&
              size > 0 && {
                height: size * unit,
                maxHeight: size * unit,
                maxWidth: size * unit,
                width: size * unit,
              },
          )}
          src={imageSrc}
          alt={title}
          title={title}
        />
      </div>
    );
  }
}

export default withStyles(
  ({ color }) => ({
    inline: {
      display: 'inline-block',
    },
    image: {
      display: 'block',
      background: color.core.neutral[6],
      color: color.base,
      objectFit: 'cover',
      overflow: 'hidden',
    },
    roundedImage: {
      borderRadius: '50%',
    },
    small: {
      height: 24,
      width: 24,
      maxHeight: 24,
      maxWidth: 24,
    },
    regular: {
      height: 48,
      width: 48,
      maxHeight: 48,
      maxWidth: 48,
    },
    large: {
      height: 96,
      width: 96,
      maxHeight: 96,
      maxWidth: 96,
    },
    macro: {
      height: 160,
      width: 160,
      maxHeight: 160,
      maxWidth: 160,
    },
  }),
  {
    passThemeProp: true,
  },
)(ProfilePhoto);
