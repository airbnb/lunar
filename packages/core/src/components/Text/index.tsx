import React from 'react';
import { mutuallyExclusiveTrueProps } from 'airbnb-prop-types';
import withStyles, { css, WithStylesProps } from '../../composers/withStyles';

const sizingProp = mutuallyExclusiveTrueProps('micro', 'small', 'large');
const emphasisProp = mutuallyExclusiveTrueProps('bold', 'light');
const stateProp = mutuallyExclusiveTrueProps('disabled', 'muted', 'inverted');
const alignProp = mutuallyExclusiveTrueProps('centerAlign', 'endAlign', 'startAlign');

export type Props = {
  /** Render the text inline instead of block. */
  baseline?: boolean;
  /** Apply bold emphasis. */
  bold?: boolean;
  /** Align the text to the center. */
  centerAlign?: boolean;
  /** The text to render. */
  children?: React.ReactNode;
  /** Mark the text as disabled. */
  disabled?: boolean;
  /** Align the text to the end. */
  endAlign?: boolean;
  /** Render the text inline-block instead of block. */
  inline?: boolean;
  /** Invert text colors. */
  inverted?: boolean;
  /** Increase font size to large. */
  large?: boolean;
  /** Apply light emphasis. */
  light?: boolean;
  /** Decrease font size to small. */
  micro?: boolean;
  /** Mark the text as muted. */
  muted?: boolean;
  /** Disable text and white space wrapping. */
  noWrap?: boolean;
  /** Preserve whitespace at the beginning and end of children. */
  preserveWhitespace?: boolean;
  /** Decrease font size to small. */
  small?: boolean;
  /** Align the text to the start. */
  startAlign?: boolean;
  /** Truncate the text with an ellipsis. */
  truncated?: boolean;
  /** Uppercase all text. */
  uppercased?: boolean;
};

/** Display a string of text with pre-defined sizing, emphasis, and state styling. */
export class Text extends React.Component<Props & WithStylesProps> {
  static propTypes = {
    bold: emphasisProp,
    centerAlign: alignProp,
    disabled: stateProp,
    endAlign: alignProp,
    inverted: stateProp,
    large: sizingProp,
    light: emphasisProp,
    micro: sizingProp,
    muted: stateProp,
    small: sizingProp,
    startAlign: alignProp,
  };

  static defaultProps = {
    baseline: false,
    bold: false,
    centerAlign: false,
    children: null,
    disabled: false,
    endAlign: false,
    inline: false,
    inverted: false,
    large: false,
    light: false,
    micro: false,
    muted: false,
    noWrap: false,
    preserveWhitespace: false,
    small: false,
    startAlign: false,
    truncated: false,
    uppercased: false,
  };

  render() {
    const {
      baseline,
      bold,
      centerAlign,
      children,
      disabled,
      endAlign,
      inline,
      inverted,
      large,
      light,
      micro,
      muted,
      noWrap,
      preserveWhitespace,
      small,
      startAlign,
      styles,
      truncated,
      uppercased,
    } = this.props;
    let Tag: 'div' | 'span' | 'small' | 'h4' = 'div';

    if (inline) {
      Tag = 'span';
    } else if (micro) {
      Tag = 'small';
    } else if (large) {
      Tag = 'h4';
    }

    return (
      <Tag
        {...css(
          styles.text,
          bold && styles.text_bold,
          disabled && styles.text_disabled,
          inline && styles.text_inline,
          baseline && styles.text_baseline,
          inverted && styles.text_inverted,
          large && styles.text_large,
          light && styles.text_light,
          micro && styles.text_micro,
          muted && styles.text_muted,
          preserveWhitespace && styles.text_preserveWhitespace,
          small && styles.text_small,
          truncated && styles.text_truncated,
          uppercased && styles.text_uppercased,
          centerAlign && styles.text_center,
          endAlign && styles.text_end,
          startAlign && styles.text_start,
          noWrap && styles.text_noWrap,
        )}
      >
        {children}
      </Tag>
    );
  }
}

export default withStyles(({ color, font, pattern }) => ({
  text: {
    ...font.textReset,
    ...font.textRegular,
    color: color.accent.text,
  },

  text_baseline: {
    display: 'inline',
  },

  text_inline: {
    display: 'inline-block',
  },

  text_preserveWhitespace: {
    whiteSpace: 'pre-wrap',
  },

  text_micro: {
    ...font.textMicro,
  },

  text_small: {
    ...font.textSmall,
  },

  text_large: {
    ...font.textLarge,
  },

  text_disabled: {
    ...pattern.disabled,
  },

  text_inverted: {
    color: color.base,
  },

  text_muted: {
    color: color.muted,
  },

  text_bold: {
    fontWeight: font.weights.semibold,
  },

  text_light: {
    fontWeight: font.weights.light,
  },

  text_truncated: {
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    whiteSpace: 'nowrap',

    '@selectors': {
      '> div': {
        display: 'inline',
      },
    },
  },

  text_uppercased: {
    textTransform: 'uppercase',
  },

  text_center: {
    textAlign: 'center',
  },

  text_end: {
    textAlign: 'right',
  },

  text_start: {
    textAlign: 'left',
  },

  text_noWrap: {
    whiteSpace: 'nowrap',
  },
}))(Text);
