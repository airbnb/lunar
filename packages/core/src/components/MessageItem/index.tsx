import React from 'react';
import { mutuallyExclusiveTrueProps } from 'airbnb-prop-types';
import withStyles, { css, WithStylesProps } from '../../composers/withStyles';
import removeFocusOnMouseUp from '../../utils/removeFocusOnMouseUp';
import ProfilePhoto from '../ProfilePhoto';
import Shimmer from '../Shimmer';
import Text from '../Text';
import Spacing from '../Spacing';
import T from '../Translate';
import toRGBA from '../../utils/toRGBA';

// add color flags here
const stripeColorTypePropType = mutuallyExclusiveTrueProps('important', 'info', 'warning');

export type Props = {
  /** Message body. */
  children: NonNullable<React.ReactNode>;
  /** Wraps title in a span to disables translation suggestions from Google Translate. */
  disableTitleTranslation?: boolean;
  /** Email display; displays that the message is from the provided email. */
  email?: React.ReactNode;
  /** Message timestamp display, e.g. "6:30 PM". */
  formattedTimestamp: string;
  /** Increase horizontal spacing. */
  horizontalSpacing?: boolean;
  /** Icon for profile photo. */
  icon?: React.ReactNode;
  /** Image URL for profile photo badge. */
  imageBadgeSrc?: string;
  /** Image description; displays on hover; defaults to `title`. */
  imageDescription?: string;
  /** Image URL. Required when using `icon`. */
  imageSrc?: string;
  /** Mark the message as important. */
  important?: boolean;
  /** Informational status (blue). */
  info?: boolean;
  /** Whether the message item is loading author data. */
  loadingAuthor?: boolean;
  /** When provided, enables the profile photo to be clicked with this callback. */
  onClickImage?: React.MouseEventHandler<HTMLButtonElement>;
  /** When provided, enables the message title to be clicked with this callback. */
  onClickTitle?: React.MouseEventHandler<HTMLButtonElement>;
  /** Whether the message item is in a sending state. */
  sending?: boolean;
  /** Text to describe the message's source, e.g. "web". */
  source?: string;
  /** Message title. */
  title: string;
  /** Text to describe the `onClickTitle`'s action; displays on hover; defaults to `title`. */
  titleClickDescription?: string;
  /** Tag to display alongside the title. */
  titleTag?: string;
  /** Increase vertical spacing. */
  verticalSpacing?: boolean;
  /** Mark the message as a warning. */
  warning?: boolean;
};

/** An individual comment within a message thread. */
export class MessageItem extends React.Component<Props & WithStylesProps> {
  static propTypes = {
    important: stripeColorTypePropType,
    info: stripeColorTypePropType,
    warning: stripeColorTypePropType,
  };

  static defaultProps = {
    disableTitleTranslation: false,
    email: null,
    horizontalSpacing: false,
    imageBadgeSrc: '',
    imageDescription: '',
    important: false,
    info: false,
    loadingAuthor: false,
    sending: false,
    source: '',
    titleClickDescription: '',
    titleTag: '',
    verticalSpacing: false,
    warning: false,
  };

  render() {
    const {
      children,
      disableTitleTranslation,
      email,
      formattedTimestamp,
      horizontalSpacing,
      icon,
      imageBadgeSrc,
      imageDescription,
      imageSrc,
      important,
      info,
      loadingAuthor,
      onClickImage,
      onClickTitle,
      sending,
      source,
      styles,
      title,
      titleClickDescription,
      titleTag,
      verticalSpacing,
      warning,
    } = this.props;
    const timestamp = source
      ? T.phrase(
          '%{time} via %{source}',
          { time: formattedTimestamp, source },
          'Timestamp and source within a message bubble',
        )
      : formattedTimestamp;

    const striped = !!(important || info || warning);
    const containerStyles = css(
      styles.container,
      horizontalSpacing && styles.container_horizontalSpacing,
      verticalSpacing && styles.container_verticalSpacing,
      striped && styles.container_withStripe,
      important && styles.container_important,
      info && styles.container_info,
      warning && styles.container_warning,
    );

    const formatedTitle = disableTitleTranslation ? (
      <span className="notranslate">{title}</span>
    ) : (
      title
    );

    if (loadingAuthor) {
      return (
        <div {...containerStyles}>
          <div {...css(styles.profilePhoto, styles.tableCell)}>
            <Shimmer width={32} height={32} radius="50%" />
          </div>

          <div {...css(styles.messageBody, styles.tableCell)}>
            <Spacing bottom={0.5}>
              <Spacing right={1} inline>
                <Shimmer width={175} height={14} />
              </Spacing>

              <Text inline small muted>
                {timestamp}
              </Text>

              {email && <Shimmer height={12} width={225} />}
            </Spacing>

            {children}
          </div>

          {sending && <div {...css(styles.sendingOverlay)} />}
        </div>
      );
    }

    let profilePhoto = null;

    if (imageSrc) {
      profilePhoto = (
        <ProfilePhoto imageSrc={imageSrc} size={4} title={imageDescription || title} />
      );
    } else if (icon) {
      profilePhoto = (
        <Spacing left={2} top={0.5}>
          {icon}
        </Spacing>
      );
    }

    const avatar = imageBadgeSrc ? (
      <div>
        {profilePhoto}

        <div {...css(styles.profileBadge)}>
          <ProfilePhoto
            inline
            imageSrc={imageBadgeSrc}
            size={2}
            title={T.phrase(
              'Profile photo badge',
              {},
              'Profile photo descriptive image text within a message bubble',
            )}
          />
        </div>
      </div>
    ) : (
      profilePhoto
    );

    return (
      <div {...containerStyles}>
        <div {...css(styles.table)}>
          <div {...css(styles.profilePhoto, styles.tableCell)}>
            {onClickImage ? (
              <button
                {...css(styles.resetButton)}
                type="button"
                title={imageDescription || title}
                onClick={onClickImage}
                onMouseUp={removeFocusOnMouseUp}
              >
                {avatar}
              </button>
            ) : (
              avatar
            )}
          </div>

          <div {...css(styles.messageBody, styles.tableCell)}>
            <Spacing bottom={0.5}>
              <span {...css(styles.messageTitle)}>
                {onClickTitle ? (
                  <button
                    {...css(styles.resetButton)}
                    type="button"
                    title={titleClickDescription || title}
                    onClick={onClickTitle}
                    onMouseUp={removeFocusOnMouseUp}
                  >
                    <Text inline bold>
                      {formatedTitle}
                    </Text>
                  </button>
                ) : (
                  <Text inline bold>
                    {formatedTitle}
                  </Text>
                )}
              </span>

              {titleTag && (
                <span {...css(styles.tag)}>
                  <Text inline micro muted>
                    {titleTag}
                  </Text>
                </span>
              )}

              <Text inline small muted>
                {timestamp}
              </Text>

              {email && (
                <Text small muted>
                  <T phrase="From: %{email}" email={email} context="Who the message is from" html />
                </Text>
              )}
            </Spacing>

            <div {...css(styles.messageBodyContent)}>{children}</div>
          </div>
        </div>

        {sending && <div {...css(styles.sendingOverlay)} />}
      </div>
    );
  }
}

export default withStyles(({ color, ui, unit, pattern }) => ({
  container: {
    position: 'relative',
    border: '1px solid transparent',
    borderRadius: ui.borderRadius,
    margin: 0,
    padding: 0,
  },

  container_horizontalSpacing: {
    paddingLeft: unit * 2,
    paddingRight: unit * 2,
  },

  container_verticalSpacing: {
    marginBottom: unit * 2,
    marginTop: unit * 2,
  },

  container_withStripe: {
    borderColor: color.accent.border,
    borderWidth: `1px 1px 1px ${unit * 0.5}px`,
    padding: `${unit * 2}px ${unit * 2}px ${unit * 2}px ${unit * 1.5 + 1}px`,
  },

  container_important: {
    backgroundColor: color.core.danger[0],
    borderLeftColor: color.core.danger[3],
  },

  container_info: {
    backgroundColor: color.accent.bg,
    borderLeftColor: color.core.primary[3],
  },

  container_warning: {
    backgroundColor: color.core.warning[0],
    borderLeftColor: color.core.warning[3],
  },

  table: {
    display: 'table',
    tableLayout: 'fixed',
    width: '100%',
  },

  tableCell: {
    display: 'table-cell',
    verticalAlign: 'top',
  },

  profilePhoto: {
    width: 4 * unit,
  },

  profileBadge: {
    position: 'absolute',
    transform: `translate(50%, ${-unit}px)`,
  },

  messageBody: {
    paddingLeft: unit,
  },

  messageBodyContent: {
    wordBreak: 'break-word',
  },

  messageTitle: {
    marginRight: unit,
    wordBreak: 'break-word',
  },

  resetButton: {
    ...pattern.resetButton,
    textAlign: 'left',
  },

  sendingOverlay: {
    position: 'absolute',
    top: '0',
    left: '0',
    width: '100%',
    height: '100%',
    pointerEvents: 'all',
    backgroundColor: toRGBA(color.core.neutral[3], 50),
    zIndex: 1,
  },

  tag: {
    border: ui.border,
    borderRadius: unit / 4,
    display: 'inline-block',
    lineHeight: 1,
    marginRight: unit,
    maxWidth: '100%',
    overflow: 'hidden',
    padding: `0 ${unit / 2}px`,
    textOverflow: 'ellipsis',
    verticalAlign: 'sub',
    whiteSpace: 'nowrap',
  },
}))(MessageItem);
