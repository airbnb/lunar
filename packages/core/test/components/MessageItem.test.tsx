import React from 'react';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import MessageItem from '../../src/components/MessageItem';
import IconCheck from '../../../icons/src/interface/IconCheck';
import ProfilePhoto from '../../src/components/ProfilePhoto';
import Text from '../../src/components/Text';

describe('<MessageItem>', () => {
  it('renders a message item', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';

    const wrapper = shallowWithStyles(
      <MessageItem formattedTimestamp={timestamp} imageSrc={url} title={title}>
        Hello world
      </MessageItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a loading author', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';

    const wrapper = shallowWithStyles(
      <MessageItem loadingAuthor formattedTimestamp={timestamp} imageSrc={url} title={title}>
        Hello world
      </MessageItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a sending overlay when message is sending', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';

    const wrapper = shallowWithStyles(
      <MessageItem sending formattedTimestamp={timestamp} imageSrc={url} title={title}>
        Hello world
      </MessageItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders an email when given', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';

    const wrapper = shallowWithStyles(
      <MessageItem
        email="foo@airbnb.com"
        formattedTimestamp={timestamp}
        imageSrc={url}
        title={title}
      >
        Hello world
      </MessageItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a source when given', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';

    const wrapper = shallowWithStyles(
      <MessageItem formattedTimestamp={timestamp} imageSrc={url} title={title} source="web">
        Hello world
      </MessageItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a title tag when given', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';
    const titleTag = 'CX';

    const wrapper = shallowWithStyles(
      <MessageItem formattedTimestamp={timestamp} imageSrc={url} title={title} titleTag={titleTag}>
        Hello world
      </MessageItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders the title with a `button` when `onClickTitle` is true', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';
    const onClickTitle = () => {};

    const wrapper = shallowWithStyles(
      <MessageItem
        formattedTimestamp={timestamp}
        imageSrc={url}
        title={title}
        onClickTitle={onClickTitle}
      >
        Hello world
      </MessageItem>,
    );

    expect(wrapper.find('button')).toHaveLength(1);
    expect(
      wrapper
        .find('button')
        .find(Text)
        .prop('children'),
    ).toBe(title);
  });

  it('renders the profile photo with a `button` when `onClickImage` is given', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';
    const onClickImage = () => {};

    const wrapper = shallowWithStyles(
      <MessageItem
        formattedTimestamp={timestamp}
        imageSrc={url}
        title={title}
        onClickImage={onClickImage}
      >
        Hello world
      </MessageItem>,
    );

    expect(
      wrapper
        .find(ProfilePhoto)
        .parent()
        .is('button'),
    ).toEqual(true);
  });

  it('renders an icon when given', () => {
    const timestamp = '11:56 AM';
    const title = 'title';
    const wrapper = shallowWithStyles(
      <MessageItem formattedTimestamp={timestamp} title={title} icon={<IconCheck decorative />}>
        Hello world
      </MessageItem>,
    );

    expect(wrapper.find(IconCheck)).toHaveLength(1);
    expect(wrapper.find(ProfilePhoto)).toHaveLength(0);
  });

  it('renders a message item with `horizontalSpacing`', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';

    const wrapper = shallowWithStyles(
      <MessageItem horizontalSpacing formattedTimestamp={timestamp} imageSrc={url} title={title}>
        Hello world
      </MessageItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a message item with `verticalSpacing`', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';

    const wrapper = shallowWithStyles(
      <MessageItem verticalSpacing formattedTimestamp={timestamp} imageSrc={url} title={title}>
        Hello world
      </MessageItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a message item with `important`', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';

    const wrapper = shallowWithStyles(
      <MessageItem important formattedTimestamp={timestamp} imageSrc={url} title={title}>
        Hello world
      </MessageItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a message item with `info` stripe', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';

    const wrapper = shallowWithStyles(
      <MessageItem info formattedTimestamp={timestamp} imageSrc={url} title={title}>
        Hello world
      </MessageItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders a message item with `warning` stripe', () => {
    const url = 'https://image.com';
    const timestamp = '11:56 AM';
    const title = 'title';

    const wrapper = shallowWithStyles(
      <MessageItem warning formattedTimestamp={timestamp} imageSrc={url} title={title}>
        Hello world
      </MessageItem>,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
