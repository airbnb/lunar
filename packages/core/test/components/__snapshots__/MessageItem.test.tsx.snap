// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageItem> renders a loading author 1`] = `
<div
  className="container_gxo12q"
>
  <div
    className="profilePhoto_h9am9q-o_O-tableCell_6x7qes"
  >
    <Shimmer
      height={32}
      radius="50%"
      width={32}
    />
  </div>
  <div
    className="messageBody_14846yz-o_O-tableCell_6x7qes"
  >
    <Spacing
      bottom={0.5}
    >
      <Spacing
        inline={true}
        right={1}
      >
        <Shimmer
          height={14}
          width={175}
        />
      </Spacing>
      <Text
        inline={true}
        muted={true}
        small={true}
      >
        11:56 AM
      </Text>
    </Spacing>
    Hello world
  </div>
</div>
`;

exports[`<MessageItem> renders a message item 1`] = `
<div
  className="container_gxo12q"
>
  <div
    className="table_cv8yzh"
  >
    <div
      className="profilePhoto_h9am9q-o_O-tableCell_6x7qes"
    >
      <ProfilePhoto
        imageSrc="https://image.com"
        size={4}
        title="title"
      />
    </div>
    <div
      className="messageBody_14846yz-o_O-tableCell_6x7qes"
    >
      <Spacing
        bottom={0.5}
      >
        <span
          className="messageTitle_ft2sf8"
        >
          <Text
            bold={true}
            inline={true}
          >
            title
          </Text>
        </span>
        <Text
          inline={true}
          muted={true}
          small={true}
        >
          11:56 AM
        </Text>
      </Spacing>
      <div
        className="messageBodyContent_1dcp06s"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`<MessageItem> renders a message item with \`horizontalSpacing\` 1`] = `
<div
  className="container_gxo12q-o_O-container_horizontalSpacing_kelmes"
>
  <div
    className="table_cv8yzh"
  >
    <div
      className="profilePhoto_h9am9q-o_O-tableCell_6x7qes"
    >
      <ProfilePhoto
        imageSrc="https://image.com"
        size={4}
        title="title"
      />
    </div>
    <div
      className="messageBody_14846yz-o_O-tableCell_6x7qes"
    >
      <Spacing
        bottom={0.5}
      >
        <span
          className="messageTitle_ft2sf8"
        >
          <Text
            bold={true}
            inline={true}
          >
            title
          </Text>
        </span>
        <Text
          inline={true}
          muted={true}
          small={true}
        >
          11:56 AM
        </Text>
      </Spacing>
      <div
        className="messageBodyContent_1dcp06s"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`<MessageItem> renders a message item with \`important\` 1`] = `
<div
  className="container_gxo12q-o_O-container_withStripe_3ecezm-o_O-container_important_zh0amp"
>
  <div
    className="table_cv8yzh"
  >
    <div
      className="profilePhoto_h9am9q-o_O-tableCell_6x7qes"
    >
      <ProfilePhoto
        imageSrc="https://image.com"
        size={4}
        title="title"
      />
    </div>
    <div
      className="messageBody_14846yz-o_O-tableCell_6x7qes"
    >
      <Spacing
        bottom={0.5}
      >
        <span
          className="messageTitle_ft2sf8"
        >
          <Text
            bold={true}
            inline={true}
          >
            title
          </Text>
        </span>
        <Text
          inline={true}
          muted={true}
          small={true}
        >
          11:56 AM
        </Text>
      </Spacing>
      <div
        className="messageBodyContent_1dcp06s"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`<MessageItem> renders a message item with \`info\` stripe 1`] = `
<div
  className="container_gxo12q-o_O-container_withStripe_3ecezm-o_O-container_info_1jxyydc"
>
  <div
    className="table_cv8yzh"
  >
    <div
      className="profilePhoto_h9am9q-o_O-tableCell_6x7qes"
    >
      <ProfilePhoto
        imageSrc="https://image.com"
        size={4}
        title="title"
      />
    </div>
    <div
      className="messageBody_14846yz-o_O-tableCell_6x7qes"
    >
      <Spacing
        bottom={0.5}
      >
        <span
          className="messageTitle_ft2sf8"
        >
          <Text
            bold={true}
            inline={true}
          >
            title
          </Text>
        </span>
        <Text
          inline={true}
          muted={true}
          small={true}
        >
          11:56 AM
        </Text>
      </Spacing>
      <div
        className="messageBodyContent_1dcp06s"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`<MessageItem> renders a message item with \`verticalSpacing\` 1`] = `
<div
  className="container_gxo12q-o_O-container_verticalSpacing_170i3mz"
>
  <div
    className="table_cv8yzh"
  >
    <div
      className="profilePhoto_h9am9q-o_O-tableCell_6x7qes"
    >
      <ProfilePhoto
        imageSrc="https://image.com"
        size={4}
        title="title"
      />
    </div>
    <div
      className="messageBody_14846yz-o_O-tableCell_6x7qes"
    >
      <Spacing
        bottom={0.5}
      >
        <span
          className="messageTitle_ft2sf8"
        >
          <Text
            bold={true}
            inline={true}
          >
            title
          </Text>
        </span>
        <Text
          inline={true}
          muted={true}
          small={true}
        >
          11:56 AM
        </Text>
      </Spacing>
      <div
        className="messageBodyContent_1dcp06s"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`<MessageItem> renders a message item with \`warning\` stripe 1`] = `
<div
  className="container_gxo12q-o_O-container_withStripe_3ecezm-o_O-container_warning_ver4t1"
>
  <div
    className="table_cv8yzh"
  >
    <div
      className="profilePhoto_h9am9q-o_O-tableCell_6x7qes"
    >
      <ProfilePhoto
        imageSrc="https://image.com"
        size={4}
        title="title"
      />
    </div>
    <div
      className="messageBody_14846yz-o_O-tableCell_6x7qes"
    >
      <Spacing
        bottom={0.5}
      >
        <span
          className="messageTitle_ft2sf8"
        >
          <Text
            bold={true}
            inline={true}
          >
            title
          </Text>
        </span>
        <Text
          inline={true}
          muted={true}
          small={true}
        >
          11:56 AM
        </Text>
      </Spacing>
      <div
        className="messageBodyContent_1dcp06s"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`<MessageItem> renders a sending overlay when message is sending 1`] = `
<div
  className="container_gxo12q"
>
  <div
    className="table_cv8yzh"
  >
    <div
      className="profilePhoto_h9am9q-o_O-tableCell_6x7qes"
    >
      <ProfilePhoto
        imageSrc="https://image.com"
        size={4}
        title="title"
      />
    </div>
    <div
      className="messageBody_14846yz-o_O-tableCell_6x7qes"
    >
      <Spacing
        bottom={0.5}
      >
        <span
          className="messageTitle_ft2sf8"
        >
          <Text
            bold={true}
            inline={true}
          >
            title
          </Text>
        </span>
        <Text
          inline={true}
          muted={true}
          small={true}
        >
          11:56 AM
        </Text>
      </Spacing>
      <div
        className="messageBodyContent_1dcp06s"
      >
        Hello world
      </div>
    </div>
  </div>
  <div
    className="sendingOverlay_f0q3k7"
  />
</div>
`;

exports[`<MessageItem> renders a source when given 1`] = `
<div
  className="container_gxo12q"
>
  <div
    className="table_cv8yzh"
  >
    <div
      className="profilePhoto_h9am9q-o_O-tableCell_6x7qes"
    >
      <ProfilePhoto
        imageSrc="https://image.com"
        size={4}
        title="title"
      />
    </div>
    <div
      className="messageBody_14846yz-o_O-tableCell_6x7qes"
    >
      <Spacing
        bottom={0.5}
      >
        <span
          className="messageTitle_ft2sf8"
        >
          <Text
            bold={true}
            inline={true}
          >
            title
          </Text>
        </span>
        <Text
          inline={true}
          muted={true}
          small={true}
        >
          11:56 AM via web
        </Text>
      </Spacing>
      <div
        className="messageBodyContent_1dcp06s"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`<MessageItem> renders a title tag when given 1`] = `
<div
  className="container_gxo12q"
>
  <div
    className="table_cv8yzh"
  >
    <div
      className="profilePhoto_h9am9q-o_O-tableCell_6x7qes"
    >
      <ProfilePhoto
        imageSrc="https://image.com"
        size={4}
        title="title"
      />
    </div>
    <div
      className="messageBody_14846yz-o_O-tableCell_6x7qes"
    >
      <Spacing
        bottom={0.5}
      >
        <span
          className="messageTitle_ft2sf8"
        >
          <Text
            bold={true}
            inline={true}
          >
            title
          </Text>
        </span>
        <span
          className="tag_5ns9qb"
        >
          <Text
            inline={true}
            micro={true}
            muted={true}
          >
            CX
          </Text>
        </span>
        <Text
          inline={true}
          muted={true}
          small={true}
        >
          11:56 AM
        </Text>
      </Spacing>
      <div
        className="messageBodyContent_1dcp06s"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;

exports[`<MessageItem> renders an email when given 1`] = `
<div
  className="container_gxo12q"
>
  <div
    className="table_cv8yzh"
  >
    <div
      className="profilePhoto_h9am9q-o_O-tableCell_6x7qes"
    >
      <ProfilePhoto
        imageSrc="https://image.com"
        size={4}
        title="title"
      />
    </div>
    <div
      className="messageBody_14846yz-o_O-tableCell_6x7qes"
    >
      <Spacing
        bottom={0.5}
      >
        <span
          className="messageTitle_ft2sf8"
        >
          <Text
            bold={true}
            inline={true}
          >
            title
          </Text>
        </span>
        <Text
          inline={true}
          muted={true}
          small={true}
        >
          11:56 AM
        </Text>
        <Text
          muted={true}
          small={true}
        >
          <Translate
            context="Who the message is from"
            email="foo@airbnb.com"
            html={true}
            phrase="From: %{email}"
          />
        </Text>
      </Spacing>
      <div
        className="messageBodyContent_1dcp06s"
      >
        Hello world
      </div>
    </div>
  </div>
</div>
`;
